src/SymmetricPositiveDefinite.jl

```diff
@@ -0,0 +1,272 @@
+@doc doc"""
+    inner(P,x,v,w)
+
+compute the inner product of `v`, `w` in the tangent space of `x` on the [`SymmetricPositiveDefinite`](@ref)
+manifold `P`, as a [`MetricManifold`](@ref) with [`LinearAffineMetric`](@ref). The formula reads
+
+```math
+( v, w)_x = \operatorname{tr}(x^{-1}\xi x^{-1}\nu ),
+```
+"""
+function inner(P::MetricManifold{SymmetricPositiveDefinite{N}, LinearAffineMetric}, x, w, v) where N
+    F = factorize(x)
+    return tr( (w / F) * (v / F ))
+end
+@doc doc"""
+    exp!(P,y,x,v)
+
+compute the exponential map from `x` with tangent vector `v` on the [`SymmetricPositiveDefinite`](@ref)
+manifold with its default metric, [`LinearAffineMetric`](@ref) and modify `y`.
+"""
+exp!(P::SymmetricPositiveDefinite{N},y,x,v) where N = exp!(MetricManifold(P,LinearAffineMetric()),y,x,v)
+@doc doc"""
+    exp!(P,y,x,v)
+
+compute the exponential map from `x` with tangent vector `v` on the [`SymmetricPositiveDefinite`](@ref)
+as a [`MetricManifold`](@ref) with [`LinearAffineMetric`](@ref) and modify `y`. The formula reads
+
+```math
+    \exp_x v = x^{\frac{1}{2}}\operatorname{Exp}(x^{-\frac{1}{2}} v x^{-\frac{1}{2}})x^{\frac{1}{2}},
+```
+where $\operatorname{Exp}$ denotes to the matrix exponential.
+"""
+function exp!(P::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric}, y, x, v) where N
+    e = eigen(Symmetric(x))
+    U = e.vectors
+    S = e.values
+    Ssqrt = Diagonal( sqrt.(S) )
+    SsqrtInv = Diagonal( 1 ./ sqrt.(S) )
+    xSqrt = U*Ssqrt*transpose(U);
+    xSqrtInv = U*SsqrtInv*transpose(U)
+    T = xSqrtInv * v * xSqrtInv
+    eig1 = eigen( T ) # numerical stabilization
+    Se = Diagonal( exp.(eig1.values) )
+    Ue = eig1.vectors
+    y = xSqrt*Ue*Se*transpose(Ue)*xSqrt
+    return y
+end
+
+@doc doc"""
+    log!(P,v,x,y)
+
+compute the logarithmic map at `x` to `y` on the [`SymmetricPositiveDefinite`](@ref)
+manifold with its default metric, [`LinearAffineMetric`](@ref) and modify `v`.
+"""
+log!(P::SymmetricPositiveDefinite{N}, v, x, y) where N = log!(MetricManifold(P,LinearAffineMetric()),y,x,v)
+@doc doc"""
+    log!(P,v,x,y)
+
+compute the exponential map from `x` to `y` on the [`SymmetricPositiveDefinite`](@ref)
+as a [`MetricManifold`](@ref) with [`LinearAffineMetric`](@ref) and modify `v`. The formula reads
+
+```math
+\log_x y = x^{\frac{1}{2}}\operatorname{Log}(x^{-\frac{1}{2}} y x^{-\frac{1}{2}})x^{\frac{1}{2}},
+```
+where $\operatorname{Log}$ denotes to the matrix logarithm.
+"""
+function log!(P::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric}, v, x, y) where N
+    e = eigen(Symmetric(x))
+    U = e.vectors
+    S = e.values
+    Ssqrt = Diagonal( sqrt.(S) )
+    SsqrtInv = Diagonal( 1 ./ sqrt.(S) )
+    xSqrt = U*Ssqrt*transpose(U)
+    xSqrtInv = U*SsqrtInv*transpose(U)
+    T = xSqrtInv * y * xSqrtInv
+    e2 = eigen( (T + transpose(T))/2 )
+    Se = Diagonal( log.(max.(e2.values,eps()) ) )
+    Ue = e2.vectors
+    v = xSqrt * Ue*Se*transpose(Ue) * xSqrt
```

Comment:
This needs broadcasting as well.

Reply:
Can you explain (or link to an explanation) what the difference is?

Reply:
It is discussed here for example: https://discourse.julialang.org/t/assignment-and-mutation/19119

Reply:
The difference is that in your case `v` (which is a new variable) points to the result of the computation on the right hand side. In particular, it is not going to overwrite the input variable `v` above, which it is supposed to do (by the exclamation mark).

I'd also suggest to cache `xSqrt * Ue` as above.